### ISSUE TITLE
Bad argument #1 to 'ipairs'

### ISSUE THREAD

**Issue by Flamme13** [mistweaverco/kulala.nvim#214]
Tell me how can I solve this problem. Everything was fine before, but now when I try to make a request, an error appears:

```
Error executing vim.schedule lua callback: ...l/share/nvim/lazy/kulala.nvim/lua/kulala/parser/init.lua:374: bad argument #1 to 'ipairs' (table expected, got nil)
```

**Comment by Flamme13** [comment]
```
return {
	"mistweaverco/kulala.nvim",
	event = "VeryLazy",
	config = function()
		require("kulala").setup({
                    default_view = "headers_body",
                })

		vim.api.nvim_set_keymap("n", "<C-\\>", ":lua require('kulala').run()<CR>", { noremap = true, silent = true })
	end,
}

```

**Comment by Flamme13** [comment]
```
==============================================================================
kulala: require("kulala.health").check()

- {kulala.nvim} version 3.7.0
- OK {curl} found

Checking formatters ~
- WARNING {text/html} formatter not found
- OK {application/json} formatter: jq .
- WARNING {application/xml} formatter not found

==============================================================================
```

**Comment by gorillamoe** [comment]
Can you try this?

```lua
return {
	"mistweaverco/kulala.nvim",
	event = "VeryLazy",
	opts = {
		winbar: true,
		default_view: "headers_body",
	},
	config = function()
		vim.api.nvim_set_keymap("n", "<C-\\>", ":lua require('kulala').run()<CR>", { noremap = true, silent = true })
	end,
}
```

**Comment by gorillamoe** [comment]
Can it be that you're on windows and do not have the latest cURL? We had a similar issue posted to Discord I think :thinking:

**Comment by gorillamoe** [comment]
What does `lua require("kulala").version()` print?

**Comment by d4be4st** [comment]
Same issue

![image](https://github.com/user-attachments/assets/18d15206-6ff8-4d5e-bc81-702bf48a6f3d)

trying

```http
GET http://localhost:3000/config
```

**Comment by gorillamoe** [comment]
Oh, Shit, I think I know the issue. I think fenced code block support might have broken it for you.

When you're in the http file, what does `:set buftype` give you?

**Comment by d4be4st** [comment]
It is not set

![image](https://github.com/user-attachments/assets/46230571-cb75-4732-b67b-9326c6173c75)

**Comment by gorillamoe** [comment]
Can you try to set it to http and run a request?

**Comment by d4be4st** [comment]
Actually it seems i just needed to set filetype

It works with 
```
filetype=http
buftype=
```

**Comment by gorillamoe** [comment]
Yeah, will push a fix. Then this is due to the newly introduced fenced code block support

**Comment by gorillamoe** [comment]
Thanks for catching this! ❤️🙏🏾

**Comment by gorillamoe** [comment]
Fix should be in main now. Re-open if the issue still persists, please.

**Comment by franckrasolo** [comment]
@gorillamoe I can confirm that this issue persists on the `main` branch at the time of writing.

**Comment by franckrasolo** [comment]
As a workaround for the time being, I've ended up adding the following keymap to the plugin configuration (LazyVim):

```lua
{
  keys = {
    { "<leader>RN", "<cmd>enew<cr><cmd>set filetype=http<cr>", desc = "HTTP scratch buffer" }
  }
}
```

or, if you'd rather stick to pure Lua:

```lua
{
  keys = {
    { mode = { "i", "n", "v" }, "<leader>RS", desc = "HTTP scratch buffer",
      function()
        vim.cmd.enew()
        vim.bo.filetype = "http"
      end,
      { silent = true, noremap = true }
    }
  }
}
```

**Comment by franckrasolo** [comment]
Ah, there's also the [scratchpad](https://kulala.mwco.app/docs/usage/public-methods#scratchpad) method!

```lua
require("kulala").scratchpad()
```

**Comment by gorillamoe** [comment]
So just to make I understand this correctly, the issue is that you open a buffer which has no file type set and the run kulala.run right?

We should catch that error and display an error message, but it won't work, because kulala runs either on http and rest files (with the file type set) or anything else within fenced code blocks.

The only exception to this is the scratchpad() function from kulala itself, which takes care setting the correct file type.

So to sum it up: kulala only runs on files with either http or rest as file type or the scratchpad. For anything else Kulala assumes you must be in a file with a fenced code block (like markdown files ..)

**Comment by franckrasolo** [comment]
> So just to make I understand this correctly, the issue is that you open a buffer which has no file type set and the run kulala.run right?

Yes, that's right. I agree about the need to handle this edge case gracefully somehow.

I've settled for these keymaps in the end and I'm unlikely to tweak them any further:

```lua
keys = {
  { "<leader>Ra", desc = "Open scratchpad",
    function()
      require("kulala").scratchpad()
    end,
    { noremap = true, silent = true }
  },
  { "<leader>Rb", desc = "Open scratchpad (empty)",
    function()
      vim.cmd.edit(require("kulala.globals").SCRATCHPAD_ID)
      vim.bo.buftype  = "nofile"
      vim.bo.filetype = "http"
    end,
    { noremap = true, silent = true }
  },
  { "<leader>Rc", desc = "Close scratchpad",
    function()
      require("kulala").close()
      vim.cmd.bdelete { args = { require("kulala.globals").SCRATCHPAD_ID }, bang = true }
    end,
    { noremap = true, silent = true }
  },
}
```

Later, I may also try executing a request from within a fenced code block.
